drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -501,9 +583,73 @@ local function occupancy_attr_handler(driver, device, ib, response)
   device:emit_event(ib.data.value == 0x01 and capabilities.motionSensor.motion.active() or capabilities.motionSensor.motion.inactive())
 end
 
+local function power_topology_handler(driver, device, ib, response)
+  -- check get_endpoints for each feature map value.
+  for name, bitmap in pairs(POWER_TOPOLOGY_FEATURE_MAP) do
+    local feature_eps = embedded_cluster_utils.get_endpoints(device, clusters.PowerTopology.ID, {feature_bitmap = bitmap})
+    if #feature_eps > 0 then
+      device:set_field(POWER_TOPOLOGY_TYPE, POWER_TOPOLOGY_FEATURE_MAP[name])
+    end
+  end
+end
+
+local function cum_energy_exported_handler(driver, device, ib, response)
+  device:set_field(TOTAL_EXPORTED_ENERGY, ib.data.elements["Energy"])
+  device:set_field(LATEST_EXPORTED_TIMESTAMP, ib.data.elements["EndTimestamp"])
+  device:emit_event(capabilities.energyMeter.energy({ value = ib.data.elements["Energy"], unit = "Wh" }))
+end
+
+local function per_energy_exported_handler(driver, device, ib, response)
+  local latest_energy_report = device:get_field(TOTAL_EXPORTED_ENERGY)
+  local summed_energy_report = latest_energy_report + ib.data.elements["Energy"]
+  device:set_field(TOTAL_EXPORTED_ENERGY, summed_energy_report)
+  device:set_field(LATEST_EXPORTED_TIMESTAMP, ib.data.elements["EndTimestamp"])
+  device:emit_event(capabilities.energyMeter.energy({ value = ib.data.elements["Energy"], unit = "Wh" }))
+end
+
+local function energy_report_handler_factory(is_cumulative_report)
+  return function(driver, device, ib, response)
+    if not device:get_field(REPORT_POLL_TIMER_IS_SET) then
+      local cumulative_eps = embedded_cluster_utils.get_endpoints(device, clusters.ElectricalEnergyMeasurement.ID,
```

Comment:
I think `is_cumulative_report` should handle all of this. If it is true, then we know we got a cumulative report, otherwise we got a periodic report

Reply:
that's true! it doesn't get used anywhere either haha